src/@types/options.ts

```diff
@@ -0,0 +1,93 @@
+import type { PartialDeep } from 'type-fest';
+import { ILevel, ILogger } from './logger';
+import { ISerializer } from './serializer';
+
+/**
+ * A function that takes in the `PrismaSessionStore` options, the `SessionData`
+ * and the `sid` for a given session and returns the TTL (time to live) in milliseconds
+ * for that particular session.
+ */
+export type TTLFactory = (
+  options: IOptions,
+  session: PartialDeep<Express.SessionData>,
+  sid: string
+) => number;
+
+/**
+ * PrismaSessionStore options to alter the way the store behaves
+ */
+export interface IOptions {
+  /**
+   * "Time to live", in ms;
+   * defines session expiration time. Defaults to session.maxAge (if set), or
+   * one day (if not set). May alternatively be set to a function, of the form
+   * `(options, session, sid) => number`.
+   */
+  ttl?: number | TTLFactory;
+
+  /**
+   * Interval, in ms, at which PrismaSessionStore will automatically remove
+   * expired sessions. Disabled by default; set to something reasonable.
+   */
+  checkPeriod?: number;
+
+  /**
+   * A flag indicating to use the session ID as the Prisma Record ID
+   *
+   * Note: If undefined and dbRecordIdFunction is also undefined then a random
+   * CUID will be used instead.
+   */
+  dbRecordIdIsSessionId?: boolean;
```

Comment:
🤔  Well session ids are created by `expressSession` I suppose you mean record ids in the database though.
In order to allow numbers we would need to change the `prisma` interface. Simply adding a type union causes a TypeError so something like a generic would be needed which would likely turn the whole `PrismaSessionStore` into a generic.

This would require the use to specific provide the type in a generic if we do not switch over to a default export (my other PR) as the type could not be inferred when the factory is called.

It is an interesting idea but I believe this should at least wait until all other PRs are complete and created into an issue that can be discussed what the value would be of making the ID generic.

Reply:
Sorry - yes, I was thinking about db record ids.

>I believe this should at least wait until all other PRs are complete and created into an issue that can be discussed what the value would be of making the ID generic.

Ok - sounds good.